### ISSUE TITLE
Version 1.5.0 Fails to start

### ISSUE THREAD

**Issue by Jsalz2000** [aerospike/aerospike-prometheus-exporter#49]
Hello!

I had recently upgraded to version 1.5.0 but when I try to run this with credentials in the fields user = and password = in ape.toml, it seems to crash for me. If I run a fresh install and do not modify the config at all it will run just fine. But as soon as I add anything inside these two fields it will fail after about 10s.

```
Mar 08 05:55:03 node3 systemd[1]: Started Aerospike Prometheus Exporter Service.
Mar 08 05:55:03 node3 aerospike-prometheus-exporter[33932]: time="2022-03-08T05:55:03Z" level=info msg="Welcome to Aerospike Prometheus Exporter v1.5.0"
Mar 08 05:55:03 node3 aerospike-prometheus-exporter[33932]: time="2022-03-08T05:55:03Z" level=info msg="Loading configuration file /etc/aerospike-prometheus-exporter/ape.toml"
Mar 08 05:55:03 node3 aerospike-prometheus-exporter[33932]: time="2022-03-08T05:55:03Z" level=info msg="Listening for Prometheus on: :9145"
Mar 08 05:55:51 node3 systemd[1]: Stopping Aerospike Prometheus Exporter Service...
Mar 08 05:55:51 node3 systemd[1]: aerospike-prometheus-exporter.service: Succeeded.
Mar 08 05:55:51 node3 systemd[1]: Stopped Aerospike Prometheus Exporter Service.
Mar 08 05:55:51 node3 systemd[1]: Started Aerospike Prometheus Exporter Service.
Mar 08 05:55:51 node3 aerospike-prometheus-exporter[34055]: time="2022-03-08T05:55:51Z" level=info msg="Welcome to Aerospike Prometheus Exporter v1.5.0"
Mar 08 05:55:51 node3 aerospike-prometheus-exporter[34055]: time="2022-03-08T05:55:51Z" level=info msg="Loading configuration file /etc/aerospike-prometheus-exporter/ape.toml"
Mar 08 05:55:51 node3 aerospike-prometheus-exporter[34055]: time="2022-03-08T05:55:51Z" level=info msg="Listening for Prometheus on: :9145"
Mar 08 05:56:00 node3 aerospike-prometheus-exporter[34055]: panic: runtime error: invalid memory address or nil pointer dereference
Mar 08 05:56:00 node3 aerospike-prometheus-exporter[34055]: [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x8f1aa6]
Mar 08 05:56:00 node3 aerospike-prometheus-exporter[34055]: goroutine 11 [running]:
Mar 08 05:56:00 node3 aerospike-prometheus-exporter[34055]: github.com/aerospike/aerospike-client-go/v5.(*sessionInfo).isValid(0x0, 0xc000207948)
Mar 08 05:56:00 node3 aerospike-prometheus-exporter[34055]:         /go/pkg/mod/github.com/aerospike/aerospike-client-go/v5@v5.7.0/login_command.go:32 +0x26
Mar 08 05:56:00 node3 aerospike-prometheus-exporter[34055]: github.com/aerospike/aerospike-client-go/v5.(*Connection).login(0xc000147360, 0xc0001ac0d0, 0xc0000ace80, 0x3c, 0x40>
Mar 08 05:56:00 node3 aerospike-prometheus-exporter[34055]:         /go/pkg/mod/github.com/aerospike/aerospike-client-go/v5@v5.7.0/connection.go:340 +0x110
Mar 08 05:56:00 node3 aerospike-prometheus-exporter[34055]: github.com/aerospike/aerospike-client-go/v5.(*Connection).Login(0xc000147360, 0xc0001ac0d0, 0xc000147360, 0x0)
Mar 08 05:56:00 node3 aerospike-prometheus-exporter[34055]:         /go/pkg/mod/github.com/aerospike/aerospike-client-go/v5@v5.7.0/connection.go:389 +0x8c
Mar 08 05:56:00 node3 aerospike-prometheus-exporter[34055]: main.newObserver.func1(0xc000151440, 0xc000151260, 0xa8e598)
Mar 08 05:56:00 node3 aerospike-prometheus-exporter[34055]:         /exporter/observer.go:125 +0x7a
Mar 08 05:56:00 node3 aerospike-prometheus-exporter[34055]: main.(*Observer).requestInfo(0xc000035480, 0x3, 0xc0000356c0, 0x4, 0x4, 0x1, 0x2, 0x4)
Mar 08 05:56:00 node3 aerospike-prometheus-exporter[34055]:         /exporter/observer.go:196 +0x33e
Mar 08 05:56:00 node3 aerospike-prometheus-exporter[34055]: main.(*Observer).refresh(0xc000035480, 0xc0001cc000, 0xc000071d40, 0xb1b9f8, 0xc000071d40)
Mar 08 05:56:00 node3 aerospike-prometheus-exporter[34055]:         /exporter/observer.go:236 +0x1a9
Mar 08 05:56:00 node3 aerospike-prometheus-exporter[34055]: main.(*Observer).Collect(0xc000035480, 0xc0001cc000)
Mar 08 05:56:00 node3 aerospike-prometheus-exporter[34055]:         /exporter/observer.go:175 +0x11b
Mar 08 05:56:00 node3 aerospike-prometheus-exporter[34055]: github.com/prometheus/client_golang/prometheus.(*Registry).Gather.func1()
Mar 08 05:56:00 node3 aerospike-prometheus-exporter[34055]:         /go/pkg/mod/github.com/prometheus/client_golang@v1.12.1/prometheus/registry.go:448 +0x154
Mar 08 05:56:00 node3 aerospike-prometheus-exporter[34055]: created by github.com/prometheus/client_golang/prometheus.(*Registry).Gather
Mar 08 05:56:00 node3 aerospike-prometheus-exporter[34055]:         /go/pkg/mod/github.com/prometheus/client_golang@v1.12.1/prometheus/registry.go:457 +0x5ce
Mar 08 05:56:00 node3 systemd[1]: aerospike-prometheus-exporter.service: Main process exited, code=exited, status=2/INVALIDARGUMENT
Mar 08 05:56:00 node3 systemd[1]: aerospike-prometheus-exporter.service: Failed with result 'exit-code'.
```

Here is the journalctl of this issue. At first you can see that I run it initially without modifying the ape.toml for about 50s and its fine, I change to add my username and password in the ape.toml, restart the instance and after 10s it fails with this error.

Maybe I am doing something wrong?

Thanks!

**Comment by spkesan** [comment]
Hey @Jsalz2000 
Thanks for reporting the issue.

Are you trying to specify credentials against the Aerospike cluster with security (or access control) disabled?
I'll try to reproduce it.

**Comment by Jsalz2000** [comment]
I had tried both. On the clusters that did not have security, it would work because I would just supply "" (default value) as username and password in the ape.toml. But in both instances (security on or off) when I put anything in these fields it would crash.

Was there any sort of dependencies change that happened between ver 1.4.1 and 1.5.0? I rolled back the version to 1.4.1 (same aerospike cluster, same ape.toml file), and then it would work fine.

**Comment by spkesan** [comment]
@Jsalz2000 @SaahilMehta 
There was a dependency upgrade in `v1.5.0` which caused logins to Aerospike nodes to fail. This is fixed in [Aerospike Prometheus Exporter v1.5.1](https://github.com/aerospike/aerospike-prometheus-exporter/releases/tag/v1.5.1).
